radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/P2PConfig.java

```diff
@@ -145,119 +146,168 @@
    */
   Duration issuedProxyCertificateValidityDuration();
 
+  /** Specifies whether the peer allow list should be used. */
+  boolean usePeerAllowList();
```

Comment:
Is there any use case where we would have `peerAllowList` non empty and this flag `false`? Otherwise, we could just use the former, I guess.

Reply:
see Davids comment above, without the additional flag the behavior when the list is empty is ambiguous

Reply:
Is it an issue? If the list is empty there is no peer to use anyway.

Reply:
Ah if the list is empty it means everything is allowed, didn't know that.

Reply:
If the default and expected behavior is to allow connection to all peers, I don't see a problem with the empty list, it doesn't mean the empty list enables this behavior, it is like that by default. On the other had, if the list isn't empty then it enables the connection only to the peers specified. I just don't like so much the idea of two configs that can be in conflict with each other.

Reply:
I think it's safer to have them separate. It prevents allowing all peers if the list is accidentally left empty (for example if some script is used to generate the config).